app/scripts/services/homePagePreference.js

```diff
@@ -0,0 +1,131 @@
+
+    function getHomePagePreferenceKey() {
+      var user = AuthService.getUser();
+      if (!user) {
+        return null;
+      }
+      return HOME_PAGE_KEY_PREFIX + user.metadata.name;
+    }
+
+    var getHomePagePreference = function() {
+      var homePagePref;
+      var key = getHomePagePreferenceKey();
+
+      if (!key) {
+        return 'catalog-home';
+      }
+
+      try {
+        homePagePref = JSON.parse(localStorage.getItem(key));
+      } catch (e) {
+        Logger.error('Could not parse homePagePref as JSON');
+        return 'catalog-home';
+      }
+
+      return _.get(homePagePref, 'type', 'catalog-home');
+    };
+
+    var getProjectOverviewPage = function() {
+      var homePagePref;
+      var key = getHomePagePreferenceKey();
+
+      if (!key) {
+        return null;
+      }
+
+      try {
+        homePagePref = JSON.parse(localStorage.getItem(key));
+      } catch (e) {
+        Logger.error('Could not parse homePagePref as JSON');
+        return null;
+      }
+
+      return (homePagePref && homePagePref.type === 'project-overview') ? homePagePref.project : null;
+    };
+
+    var verifyProjectOverviewPage = function(projectOverviewPage) {
+      var deferred = $q.defer();
+
+      ProjectsService.get(projectOverviewPage).then(function() {
+        deferred.resolve(true);
+      }, function() {
+        deferred.resolve(false);
+      });
+
+      return deferred.promise;
+    };
+
+    var clear = function() {
+      var key = getHomePagePreferenceKey();
+      if (!key) {
+        return;
+      }
+      localStorage.removeItem(key);
+    };
+
+    var notifyInvalidProjectHomePage = function(projectName) {
+      clear();
+      NotificationsService.addNotification({
+        id: "invalid-home-page-preference",
+        type: "warning",
+        message: "Home page project not found.",
+        details: "Project " + projectName + " no longer exists or you do not have access to it.",
+        links: [{
+          href: "",
+          label: "Set Home Page",
+          onClick: function() {
+            $uibModal.open({
+              animation: true,
+              templateUrl: 'views/modals/set-home-page-modal.html',
+              controller: 'SetHomePageModalController'
+            });
+            return true;
+          }
+        }]
+      });
+    };
+
+    var gotoHomePage = function() {
+      var homePagePref = this.getHomePagePreference();
+      if (homePagePref === "project-overview") {
+        var homeProjectPage = this.getProjectOverviewPage();
+        this.verifyProjectOverviewPage(homeProjectPage).then(function (valid) {
+          if (valid) {
+            Navigate.toProjectOverview(homeProjectPage);
+          } else {
+            notifyInvalidProjectHomePage(homeProjectPage);
+            Navigate.toProjectList();
+          }
+        });
+      } else if (homePagePref === "project-list") {
+        // navigate on next digest cycle
+        $timeout(function(){
```

Comment:
Why is the timeout needed? Because we're calling from a `run()` block? I'd rather not have this in the service, but have it in the run block. But I'm guessing we're be able to remove this if we switch to `redirectTo`